database/oracle/oracle.go

```diff
@@ -0,0 +1,435 @@
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return &database.Error{OrigErr: err, Err: "transaction commit failed"}
+	}
+
+	return nil
+}
+
+func (ora *Oracle) Version() (version int, dirty bool, err error) {
+	query := "SELECT VERSION, DIRTY FROM " + ora.config.MigrationsTable + " WHERE ROWNUM = 1 ORDER BY VERSION desc"
+	err = ora.conn.QueryRowContext(context.Background(), query).Scan(&version, &dirty)
+	switch {
+	case err == sql.ErrNoRows:
+		return database.NilVersion, false, nil
+
+	case err != nil:
+		if _, ok := err.(OraErr); ok {
+			return database.NilVersion, false, nil
+		}
+		return 0, false, &database.Error{OrigErr: err, Query: []byte(query)}
+
+	default:
+		return version, dirty, nil
+	}
+}
+
+func (ora *Oracle) Drop() (err error) {
+	// select all tables in current schema
+	query := fmt.Sprintf(`SELECT TABLE_NAME FROM USER_TABLES`)
+	tables, err := ora.conn.QueryContext(context.Background(), query)
+	if err != nil {
+		return &database.Error{OrigErr: err, Query: []byte(query)}
+	}
+	defer func() {
+		if errClose := tables.Close(); errClose != nil {
+			err = multierror.Append(err, errClose)
+		}
+	}()
+
+	// delete one table after another
+	tableNames := make([]string, 0)
+	for tables.Next() {
+		var tableName string
+		if err := tables.Scan(&tableName); err != nil {
+			return err
+		}
+		if len(tableName) > 0 {
+			tableNames = append(tableNames, tableName)
+		}
+	}
+
+	query = `
+BEGIN
+   EXECUTE IMMEDIATE 'DROP TABLE %s';
+EXCEPTION
+   WHEN OTHERS THEN
+      IF SQLCODE != -942 THEN
+         RAISE;
+      END IF;
+END;
+`
+	if len(tableNames) > 0 {
+		// delete one by one ...
+		for _, t := range tableNames {
+			if _, err := ora.conn.ExecContext(context.Background(), fmt.Sprintf(query, t)); err != nil {
+				return &database.Error{OrigErr: err, Query: []byte(query)}
+			}
+		}
+	}
+
+	return nil
+}
+
+// ensureVersionTable checks if versions table exists and, if not, creates it.
+// Note that this function locks the database, which deviates from the usual
+// convention of "caller locks" in the Postgres type.
+func (ora *Oracle) ensureVersionTable() (err error) {
+	if err = ora.Lock(); err != nil {
+		return err
+	}
+
+	defer func() {
+		if e := ora.Unlock(); e != nil {
+			if err == nil {
+				err = e
+			} else {
+				err = multierror.Append(err, e)
+			}
+		}
+	}()
+
+	query := `
+declare
+v_sql LONG;
+begin
+
+v_sql:='create table %s
+  (
+  VERSION NUMBER(20) NOT NULL PRIMARY KEY,
+  DIRTY NUMBER(1) NOT NULL
+  )';
+execute immediate v_sql;
+
+EXCEPTION
+    WHEN OTHERS THEN
+      IF SQLCODE = -955 THEN
+        NULL; -- suppresses ORA-00955 exception
+      ELSE
+         RAISE;
+      END IF;
+END;
+`
+	if _, err = ora.conn.ExecContext(context.Background(), fmt.Sprintf(query, ora.config.MigrationsTable)); err != nil {
+		return &database.Error{OrigErr: err, Query: []byte(query)}
+	}
+
+	return nil
+}
+
+func parseStatements(rd io.Reader, plsqlStatementSeparator string) ([]string, error) {
```

Comment:
This should be optional. e.g. migrations should be treated as blobs

See: https://github.com/golang-migrate/migrate/blob/master/MIGRATIONS.md#migration-content-format
You can use `x-multi-statement` and `x-multi-statement-plsql` as an options to enable multiple statements. See the cassandra driver for an example

Reply:
In my use case, I hope I can write the migration file like Postgres do, i.e, write multi-statements in a single file,  so I prefer to enable multi-statements support by default, and meanwhile, the user can disable multi-statements via `x-disable-multi-statements=true`